pilot/pkg/serviceregistry/cloudfoundry/servicediscovery.go

```diff
@@ -99,6 +99,12 @@ func (sd *ServiceDiscovery) GetService(hostname model.Hostname) (*model.Service,
 	return nil, nil
 }
 
+// GetServiceNamespace retrieves namespace of a service if it exists. Currently it's not implemented
+// and only returns empty string.
+func (sd *ServiceDiscovery) GetServiceNamespace(service *model.Service) (string, error) {
+	return "", errors.New("not implemented")
+}
+
```

Comment:
return default, nil. There is no need to cause errors in other platforms. Best case, these platforms will also get some RBAC or other stuff.